lib/chef/knife/data_bag_create.rb

```diff
@@ -49,13 +49,15 @@ def run
           exit(1)
         end
 
-        # create the data bag
+        # Verify if the data bag exists
         begin
+          rest.get("data/#{@data_bag_name}")
+          ui.info("Data bag #{@data_bag_name} already exists")
```

Comment:
Rather than do exception handling, you may want to just store the response and use something like:

```ruby
case response.code
when 409
   ui.info("Data bag #{@data_bag_name} already exists")
when 404
  rest.post("data", { "name" => @data_bag_name })
  ui.info("Created data_bag[#{@data_bag_name}]")
when >= 500
  raise Net::HTTPServerException
end 
```